boot/boot_test.go

```diff
@@ -315,6 +312,34 @@ func setupUC20Bootenv(c *C, bl bootloader.Bootloader, opts *bootenv20Setup) (res
 	}
 }
 
+func setupUC20MockBootloaderEnv(c *C, bl bootloader.Bootloader, opts *bootenv20Setup, cleanups *[]func()) {
+	origEnv, err := bl.GetBootVars("snap_kernel", "snap_try_kernel")
+	c.Assert(err, IsNil)
+	m := make(map[string]string, 2)
+	if opts.kern != nil {
+		m["snap_kernel"] = opts.kern.Filename()
+	} else {
+		m["snap_kernel"] = ""
+	}
+
+	if opts.tryKern != nil {
+		m["snap_try_kernel"] = opts.tryKern.Filename()
+	} else {
+		m["snap_try_kernel"] = ""
+	}
+
+	err = bl.SetBootVars(m)
+	c.Assert(err, IsNil)
+
+	// don't count any calls to SetBootVars made thus far
+	//bl.SetBootVarsCalls = 0
```

Comment:
is this not needed here? I don't remember all of the details but I seem to recall that some calls to SetBootVars are done while setting up tests and we don't want to include those when we test things like how many calls we do to SetBootVars

Reply:
It is needed, but we need to do in in `setupUC20Bootenv()` as the concrete type is needed. This was some left-over that I forgot to remove, thanks for pointing out.